### PR TITLE
Add queries and mutations related to authenticated user/teacher

### PR DESCRIPTION
**Pull request by chicobentojr**
- [x] Ability to see information about the authenticated user (#3)
- [x] Ability to list questions from an authenticated teacher (#4)
- [x] Ability to check if a questions is subjective or not (without manually checking options) (#5)
- [x] Ability to check if a user is a teacher or not (without manually checking subjects) (#6)
- [x] Ability to create Subject as a teacher

#2

### CODE REVIEW

File: management/decorators.py [hunk 1]
```diff
@@ -29,3 +28,12 @@ def wrapper(context, *args, **kwargs):
             return f(*args, **kwargs)
         raise Exception('You do not have permission to perform this action')
     return wrapper
+
+def teacher_required(f):
+    @wraps(f)
+    @context(f)
+    def wrapper(context, *args, **kwargs):
+        if context.user.is_anonymous or not context.user.is_teacher:
```
Comment (felipemfp) [inline management/decorators.py:36]: Maybe check if the user is active?

File: questions/queries.py [hunk 2]
```diff
@@ -20,9 +20,16 @@ class QuestionQuery(graphene.ObjectType):
 
     question = graphene.Field(QuestionType, id=graphene.UUID())
     questions = graphene.List(QuestionType)
+    my_questions = graphene.List(QuestionType)
```
Comment (felipemfp) [inline questions/queries.py:23]: I think we should remove `questions` and rename `my_questions` to `questions`

**Review by felipemfp — APPROVED** [review]
Take a look in my comments and feel free to choose between refactor this PR or include in the next one. :+1:

File: management/decorators.py (hunk 1 above)
Comment (chicobentojr) [reply management/decorators.py:36]: I will do this in the next PR

File: questions/queries.py (hunk 2 above)
Comment (chicobentojr) [reply questions/queries.py:23]: I will do this in the next PR

**Pull request merged**